projects/angular/src/lib/maskito.directive.ts

```diff
@@ -30,16 +30,16 @@ export class MaskitoDirective implements OnDestroy, OnChanges {
 
     @Input()
     maskitoElement: MaskitoElementPredicate = e =>
-        e.querySelector('input,textarea') ||
+        e.querySelector<HTMLInputElement | HTMLTextAreaElement>('input,textarea') ||
         (e as HTMLInputElement | HTMLTextAreaElement);
 
-    ngOnChanges(): void {
+    async ngOnChanges(): Promise<void> {
         this.maskedElement?.destroy();
-        this.ngZone.runOutsideAngular(() => {
-            this.maskedElement = new Maskito(
-                this.maskitoElement(this.elementRef.nativeElement),
-                this.maskito,
-            );
+
+        await this.ngZone.runOutsideAngular(async () => {
+            const predicate = await this.maskitoElement(this.elementRef.nativeElement);
```

Comment:
I think this should be outside ngZone wrap:
```ts
const element = await this.maskitoElement(this.elementRef.nativeElement);

this.ngZone.runOutsideAngular(() => {
    this.maskedElement = new Maskito(predicate, this.maskito);
});
```
This way asynchronous part is more pinpoint. If the predicate is synchronous, does this introduce a microtask delay to the processing or does it work synchronously? I don't really remember the specs on this, but I think it is synchronous in that case 🤔